### PR TITLE
Revised watch replay

### PR DESCRIPTION
**Pull request by lukew3**
Made the changes requested in pr #1089, which can now be closed. Replay looks just like it would during a test. May be some minor edge cases that break it right now, but most use cases are covered.

### CODE REVIEW

**Comment by lukew3** [comment]
[Demo](https://youtu.be/Abot8lzKb1I) Sorry if my typing seem incontinuous, I was trying to think about edge cases to show while I was typing.

**Comment by lukew3** [comment]
[Here's a more realistic usage demo](https://youtu.be/0lWeafiMhq0)

**Comment by Miodec** [comment]
Things i found:
 - Cant pause the replay
 - Pressing play makes the whole result jump by like  pixels.
 - Pressing play over and over again makes the result jitter.
 - Ctrl + backspace breaks the replay

File: src/js/script.js [hunk 1]
```diff
@@ -1,3 +1,9 @@
+import {
+  replayGetWordsList,
+  clearReplayData,
+  startReplayTracking,
+  stopReplayTracking,
+} from "./replay.js";
```
Comment (Miodec) [inline src/js/script.js:6]: Also in terms of using modules - I think I would prefer the syntax `Replay.getWordsList`, `Replay.clearData` like the other modules.

**Comment by Miodec** [comment]
In terms of getting data into the module - I should be rewriting test logic into a module so it should get a bit easier in that.

**Comment by lukew3** [comment]
I fixed your second and third issue after my initial pr. I can add a pause button if you'd like. Are you saying ctrl+backspace breaks it during playback or if you type it during the actual test?

**Comment by Miodec** [comment]
> I fixed your second and third issue after my initial pr. I can add a pause button if you'd like. Are you saying ctrl+backspace breaks it during playback or if you type it during the actual test?

If you ctrl backspace during the test.

**Comment by UnrealApex** [comment]
> [Here's a more realistic usage demo](https://youtu.be/0lWeafiMhq0)

@lukew3 
Another option you can use besides uploading videos to Youtube would be attaching a video file to your comment. See [File attachments on issues and pull requests](https://docs.github.com/en/github/managing-your-work-on-github/file-attachments-on-issues-and-pull-requests#:~:text=To%20attach%20a%20file%20to,images%20directly%20into%20the%20box.) for more clarification.

**Comment by lukew3** [comment]
@UnrealApex Didn't know you could do that with videos. I'll do that next time, thanks for the tip!

**Comment by Miodec** [comment]
Any updates? Need help?

**Comment by lukew3** [comment]
@Miodec Can't really work on it right now because of school. Should be able to work on it after May 12th. I don't mind if you want to help out in the meantime.

**Comment by WolfTypist** [comment]
Dang man this is awesome! How can i add the replay feature to my monkey type on my end like you did in the YouTube video? Or do i have to be a developer or something?

**Comment by UnrealApex** [comment]
@WolfTypist 

You can fork @lukew3 's fork of Monkeytype, but I'm sure it's not ready for use yet.

**Comment by lukew3** [comment]
Update: I've got everything working on my branch, but the project has changed a lot since I was here last, so I'm just working on merge errors now.

**Comment by lukew3** [comment]
https://user-images.githubusercontent.com/47042841/117087541-c83c8b80-ad1d-11eb-929d-bb9a04a15410.mp4

**Comment by UnrealApex** [comment]
@lukew3 
It would be nice to see the time elapsed during the test while watching the replay. What do you think?

**Comment by lukew3** [comment]
@UnrealApex 
>  It would be nice to see the time elapsed during the test while watching the replay. What do you think?
Maybe, I'm not exactly sure where it would go though. I think that a better idea would be to have a progress line on the graph that moves as the replay progresses. That way the user could see the time and track their performance with the graph. I'm not sure how hard it is to mess with the graph though.

**Comment by UnrealApex** [comment]
Sounds good.

**Comment by lukew3** [comment]
@Miodec I think this PR is ready.

**Comment by UnrealApex** [comment]
You could put it under the regular time spot.

**Comment by UnrealApex** [comment]
@WolfTypist 
> Dang man this is awesome! How can i add the replay feature to my monkey type on my end like you did in the YouTube video? Or do i have to be a developer or something?

There are several amazing free resource on the internet you can use to learn web development or programming in general. Some websites I would suggest are [Freecodecamp](https://www.freecodecamp.org/), [Code.org](https://code.org/), [Khan Academy](https://www.khanacademy.org/computing), [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Learn), [W3Schools](https://www.w3schools.com/), [Guru99](https://www.guru99.com/), [TutorialsPoint](https://www.tutorialspoint.com/index.htm), [pythonrprogramming.net](https://pythonprogramming.net/discordpy-basic-bot-tutorial-introduction/), [MIT's Missing Semester](https://missing.csail.mit.edu/)and more. [Heres a video to set you on your path to becoming a developer](https://coder-coder.com/learn-web-development/).

**Comment by Miodec** [comment]
Whats up with all the whitespace changes?

**Comment by lukew3** [comment]
> Whats up with all the whitespace changes?

@Miodec What do you mean?

**Comment by Miodec** [comment]
image

**Comment by lukew3** [comment]
Should I just make a new pr with only the necessary changes?

**Comment by Miodec** [comment]
Yeah sure.
